internal/splunk/httprequest.go

```diff
@@ -29,6 +32,23 @@ func HandleHTTPCode(resp *http.Response) error {
 		resp.StatusCode,
 		http.StatusText(resp.StatusCode))
 
+	// Check if there is any error text returned by Splunk that we can append to the error message
+	if resp.Body != nil {
+		var jsonResponse map[string]any
+		bodyString, _ := io.ReadAll(resp.Body)
+		resp.Body = io.NopCloser(bytes.NewBuffer(bodyString))
+		unmarshalError := json.Unmarshal(bodyString, &jsonResponse)
+
+		if unmarshalError == nil {
+			responseErrorValue, ok := jsonResponse["text"]
+			if ok {
+				err = multierr.Append(err, fmt.Errorf("reason: %v", responseErrorValue))
+			} else {
+				err = multierr.Append(err, fmt.Errorf("no error message from Splunk found"))
```

Comment:
reviewing this - this is going to sound pretty mysterious, any chance we can not add an error then?

Reply:
Yeah, we could just not add anything, that would be ok. Since we are already doing that when `if resp.Body != nil` we could do the same here. Or we could add a default error in both cases.